### PR TITLE
Enhance Helm chart deletion Logic in cleanup.sh Script  

### PR DESCRIPTION
**Pull request by anpanigr**
Modified the clean up scripts to loop thru the NameSpace which has active Helm Deployment for the deletion of Helm Deployment.  The new logic will take care of  both Helm 2.x/3.x

### CODE REVIEW

File: src/integration-tests/bash/cleanup.sh [hunk 1]
```diff
@@ -523,32 +523,38 @@ if [ -x "$(command -v helm)" ]; then
...
+     echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+   fi
+   done
+  fi
+
+  if [ "$HELM_VERSION" == "V3" ]; then
+   namespaces=`helm list --all-namespaces | grep -v NAME | awk '{print $2}'`
+   for ns in $namespaces
+   do 
+     echo "Looking for helm lists in namesapce $ns"
```
Comment (markxnelson) [inline src/integration-tests/bash/cleanup.sh]: The loop is still here

File: src/integration-tests/bash/cleanup.sh [hunk 2]
```diff
@@ -523,32 +523,38 @@ if [ -x "$(command -v helm)" ]; then
...
-     helm list --short --namespace $ns | while read helm_name; do
-       if [ "$HELM_VERSION" == "V2" ]; then
-         echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+     echo @@ `timestamp` Info: DRYRUN: helm delete --purge  $helm_name
+   fi
+   done
+  fi
+
+  if [ "$HELM_VERSION" == "V3" ]; then
+   namespaces=`helm list --all-namespaces | grep -v NAME | awk '{print $2}'`
```
Comment (tbarnes-us) [inline src/integration-tests/bash/cleanup.sh]: I suspect this should be:      

helm list --all-namespaces | grep -v NAME | awk '{print $2}' | sort -u

We only want to iterate through each namespace once.

File: src/integration-tests/bash/cleanup.sh (hunk 1 above)
Comment (anpanigr) [reply src/integration-tests/bash/cleanup.sh]: namespaces variable will store only the namesapce(s) that has a helm deploymet

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: The loop here is OK IMO, as long as we only iterate through each NS once per my previous comment.

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: But if you want to further optimize, iterate through the output of the first all namespace command in a single "big loop" that parses its lines, which which parses out the ns from the line and parses out the helm name from the same line.

File: src/integration-tests/bash/cleanup.sh (hunk 2 above)
Comment (anpanigr) [reply src/integration-tests/bash/cleanup.sh]: nice suggestion. will implement change in next commit

Comment (tbarnes-us) [reply src/integration-tests/bash/cleanup.sh]: i am assuming that the above command does not yield a ns unless there's at least one helm release in the ns

**Comment by tbarnes-us** [comment]
> The list gives you namespace and release, awk print them both and delete

This is more efficient than my 'sort -u' suggestion above.

**Comment by anpanigr** [comment]
I tried the following 
helm list --all-namespaces | grep -v NAME | awk '{system("helm uninstall $1 --namespace $2 ")}'
But it throws helm usage  ( Error: flag needs an argument: --namespace )
Am I doing something wrong ?  I am not a awk expert

**Comment by markxnelson** [comment]
2020-03-30 20:12:33:~ 
$ helm version
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}

2020-03-30 20:12:36:~ 
$ helm list --all-namespaces
NAME                            	NAMESPACE                     	REVISION	UPDATED                                	STATUS	CHART                  	APP VERSION
wls-operator-bobs-books-binding1	verrazzano-bobs-books-binding1	1       	2020-03-30 20:16:06.628931236 +0000 UTC	failed	weblogic-operator-2.5.0

**Comment by markxnelson** [comment]
what version of helm do you have? i have 3.1.2

**Comment by markxnelson** [comment]
$ helm list --all-namespaces | grep -v NAME | awk ' { print "helm uninstall " $1 " --namespace " $2 }' 
helm uninstall wls-operator-bobs-books-binding1 --namespace verrazzano-bobs-books-binding1

**Comment by anpanigr** [comment]
I am using following helm 
version.BuildInfo{Version:"v3.0.2", GitCommit:"19e47ee3283ae98139d98460de796c1be1e3975f", GitTreeState:"clean", GoVersion:"go1.13.5"}

**Comment by markxnelson** [comment]
$ helm list --all-namespaces | grep -v NAME | awk ' { system( "helm uninstall " $1 " --namespace " $2) }' 
release "wls-operator-bobs-books-binding1" uninstalled

**Comment by markxnelson** [comment]
ok let me know when you are ready @anpanigr 
i will run a jenkins job to confirm it works, and then merge it tonight before the nightly test run

**Comment by tbarnes-us** [comment]
> $ helm list --all-namespaces | grep -v NAME | awk ' { system( "helm uninstall " $1 " --namespace " $2) }'

Here's a modification of the above that's a bit harder to read (code looks more complex), but should be easier to debug if something goes wrong at runtime:

awk ' { system( "echo -n About to helm uninstall " $1 " --namespace " $2 ": ; helm uninstall " $1 " --namespace " $2 " ) }' 

It might be that the set -x that's in there already will do the above.  But if it doesn't...

**Comment by tbarnes-us** [comment]
> > $ helm list --all-namespaces | grep -v NAME | awk ' { system( "helm uninstall " $1 " --namespace " $2) }'
> 
> Here's a modification of the above that's a bit harder to read (code looks more complex), but should be easier to debug if something goes wrong at runtime:
> 
> awk ' { system( "echo -n About to helm uninstall " $1 " --namespace " $2 ": ; helm uninstall " $1 " --namespace " $2 " ) }'
> 
> It might be that the set -x that's in there already will provide sufficient debug output. But if it doesn't...

Or even better:

awk ' { system( "set -x ; helm uninstall " $1 " --namespace " $2 " ) }'

(The set -x scopes inside the system call.)

File: src/integration-tests/bash/cleanup.sh [hunk 3]
```diff
@@ -537,23 +537,15 @@ if [ -x "$(command -v helm)" ]; then
...
-        )
-       else 
-         echo @@ `timestamp` Info: DRYRUN: helm uninstall $helm_name -n $ns
-       fi
-     done
-   done
+    if [ ! "$DRY_RUN" = "true" ]; then
+    (
+      set -x
+      helm list --all-namespaces | grep -v NAME | awk '{system("helm uninstall " $1 " --namespace " $2)}'
```
Comment (tbarnes-us) [inline src/integration-tests/bash/cleanup.sh]: Change "helm uninstall" to "set -x ; helm uninstall".  In my testing it turns out that the outer scoped set -x is suppressed by teh the system call scope, so you need one inside the system call scope too.

**Comment by markxnelson** [comment]
@rjeberhard cherry picked this into frozen branch model-in-image as this is an important fix for nightly tests

**Pull request merged**